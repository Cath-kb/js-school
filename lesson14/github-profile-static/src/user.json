{
  "login": "Cath-kb",
  "id": 2164575,
  "node_id": "MDQ6VXNlcjIxNjQ1NzU=",
  "avatar_url": "https://avatars3.githubusercontent.com/u/2164575?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/Cath-kb",
  "html_url": "https://github.com/Cath-kb",
  "followers_url": "https://api.github.com/users/Cath-kb/followers",
  "following_url": "https://api.github.com/users/Cath-kb/following{/other_user}",
  "gists_url": "https://api.github.com/users/Cath-kb/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/Cath-kb/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/Cath-kb/subscriptions",
  "organizations_url": "https://api.github.com/users/Cath-kb/orgs",
  "repos_url": "https://api.github.com/users/Cath-kb/repos",
  "events_url": "https://api.github.com/users/Cath-kb/events{/privacy}",
  "received_events_url": "https://api.github.com/users/Cath-kb/received_events",
  "type": "User",
  "site_admin": false,
  "name": "Catherine Chernysheva",
  "company": null,
  "blog": "https://www.linkedin.com/in/cathkb",
  "location": "Kyiv, Ukraine",
  "email": null,
  "hireable": null,
  "bio": null,
  "public_repos": 21,
  "public_gists": 1,
  "followers": 4,
  "following": 0,
  "created_at": "2012-08-16T17:44:34Z",
  "updated_at": "2018-08-05T08:14:18Z"
}
