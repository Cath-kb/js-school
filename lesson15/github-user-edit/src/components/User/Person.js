import React from "react";

class Person extends React.Component {
  onClick = (e) => {
    e.preventDefault();
    this.props.onClick();
  };

  render() {
    const { person } = this.props;

    return (
      <React.Fragment>
        <h2>
          <a href="" title="Edit profile" onClick={this.onClick} className="link-styled">
            {person.name || 'no name'}
          </a>
        </h2>
        <h3>
          <a href={person.origin} target="_blank" className="link-unstyled" title="see on github">{person.login}</a>
        </h3>
        {person.company && <p>{person.company}</p>}
        {person.location && <p>{person.location}</p>}
        <h4><strong>Bio:</strong></h4>
        <p>{person.bio || 'no bio entered'}</p>
      </React.Fragment>
    );
  }
}

export default Person;
