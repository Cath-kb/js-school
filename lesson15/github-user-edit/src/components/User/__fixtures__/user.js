const user = {
  avatar: {
    url: 'https://avatars3.githubusercontent.com/u/2164575?v=4',
  },
  person: {
    name: 'John Doe',
    login: 'joe',
    company: '',
    location: 'Aldebaran',
  },
  statistics: {
    repos: 5,
    following: 0,
    followers: 5,
  }
};

export default user;