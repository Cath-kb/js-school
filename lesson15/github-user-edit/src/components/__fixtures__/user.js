const user = {
  avatar: {
    url: 'https://avatars3.githubusercontent.com/u/2164575?v=4',
  },
  person: {
    name: 'John Doe',
    login: 'joe',
    company: '',
    location: 'Aldebaran',
    origin: 'https://github.com/joe',
  },
  statistics: {
    repos: 5,
    following: 0,
    followers: 5,
  }
};

export default user;