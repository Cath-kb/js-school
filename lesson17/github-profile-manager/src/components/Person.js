import React from "react";
import {Link} from "react-router-dom";

import UserProfileContext from "../model/UserProfileContext";

class Person extends React.Component {
  onClick = (e) => {
    e.preventDefault();
    this.props.onClick();
  };

  render() {
    return (
      <UserProfileContext.Consumer>
        {
          ({ person }) => {
            const name = person.name || 'no name';

            return (
              <React.Fragment>
                <img
                  className="img-fluid mb-4 rounded"
                  src={person.avatar.url}
                  alt="User Avatar"
                />
                <h2>
                  {
                    this.props.active
                      ? <Link to={`/${person.login}/edit`} title="Edit profile" className="link-styled">{name}</Link>
                      : name
                  }
                </h2>
                <h3>
                  <a href={person.origin} target="_blank" className="link-unstyled" title="see on github">{person.login}</a>
                </h3>
                {person.company && <p>{person.company}</p>}
                {person.location && <p>{person.location}</p>}
                <h4><strong>Bio:</strong></h4>
                <p>{person.bio || 'no bio entered'}</p>
              </React.Fragment>
          )}}
      </UserProfileContext.Consumer>
    );
  }
}

export default Person;
