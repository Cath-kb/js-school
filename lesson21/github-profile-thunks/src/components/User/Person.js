import React from "react";

const Person = ({person}) => (
  <div>
    <h2>{person.name}</h2>
    <h3>{person.login}</h3>
    {person.company && <p>{person.company}</p>}
    {person.location && <p>{person.location}</p>}
  </div>
);

export default Person;
