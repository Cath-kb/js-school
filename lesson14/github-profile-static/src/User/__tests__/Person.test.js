import React from 'react';
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer";
import Person from './../Person';
import user from './../__fixtures__/user';

const { person } = user;

describe('Person', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Person person={person} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be rendered without errors', () => {
    const html = renderer.create(<Person person={person} />).toJSON();
    expect(html).toMatchSnapshot();
  });
});
