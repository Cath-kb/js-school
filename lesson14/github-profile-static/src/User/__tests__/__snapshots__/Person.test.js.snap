// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person should be rendered without errors 1`] = `
<div>
  <h2>
    John Doe
  </h2>
  <h3>
    joe
  </h3>
  
  <p>
    Aldebaran
  </p>
</div>
`;
